### PR TITLE
enhance: Refine error message for parsing expr

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #30710

### CODE REVIEW

File: internal/parser/planparserv2/parser_visitor.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (v *ParserVisitor) translateIdentifier(identifier string) (*ExprWithType, e
 		nestedPath = append(nestedPath, identifier)
 	}
 	if typeutil.IsJSONType(field.DataType) && len(nestedPath) == 0 {
-		return nil, fmt.Errorf("can not comparisons jsonField directly")
+		return nil, fmt.Errorf("cannot directly compare JSON dataType with other dataType")
```
Comment (xiaofan-luan) [inline internal/parser/planparserv2/parser_visitor.go:40]: print what data it is (and if possible show more details abou the rexpression)

**Comment by xiaocai2333** [comment]
rerun ut

File: tests/python_client/testcases/test_search.py [hunk 2]
```diff
@@ -494,9 +494,9 @@ def test_search_with_expression_join_two_fields(self, expression):
         collection_w.search(vectors[:default_nq], default_search_field,
                             default_search_params, nb, expression,
                             check_task=CheckTasks.err_res,
-                            check_items={"err_code": 1,
+                            check_items={"err_code": 1100,
```
Comment (czs007) [inline tests/python_client/testcases/test_search.py:497]: 2200?

File: tests/python_client/testcases/test_search.py [hunk 3]
```diff
@@ -575,7 +575,7 @@ def test_search_with_expression_invalid_like(self, expression):
         search_res, _ = collection_w.search(vectors[:default_nq], default_search_field,
                                             default_search_params, default_limit, expression,
                                             check_task=CheckTasks.err_res,
-                                            check_items={"err_code": 1,
+                                            check_items={"err_code": 1100,
```
Comment (czs007) [inline tests/python_client/testcases/test_search.py:578]: 2200

File: tests/python_client/testcases/test_search.py [hunk 4]
```diff
@@ -627,9 +627,8 @@ def test_search_with_expression_invalid_array_two(self):
 
         # 2. search
         expression = "int32_array[0] - 1 < 1"
-        error = {ct.err_code: 65535,
-                 ct.err_msg: f"failed to create query plan: cannot parse expression: {expression}, "
-                             f"error: LessThan is not supported in execution backend"}
+        error = {ct.err_code: 1100,
```
Comment (czs007) [inline tests/python_client/testcases/test_search.py:630]: 2201

File: tests/python_client/testcases/test_search.py [hunk 5]
```diff
@@ -9293,8 +9292,8 @@ def test_search_json_expression_object(self):
                             default_search_params, default_limit,
                             json_search_exp,
                             check_task=CheckTasks.err_res,
-                            check_items={ct.err_code: 1,
-                                         ct.err_msg: "can not comparisons jsonField directly"})
+                            check_items={ct.err_code: 2200,
```
Comment (czs007) [inline tests/python_client/testcases/test_search.py:9295]: 2201

File: tests/python_client/testcases/test_query.py [hunk 6]
```diff
@@ -2511,9 +2511,8 @@ def test_query_string_with_invalid_prefix_expr(self):
         expression = 'float like "0%"'
         collection_w.query(expression,
                            check_task=CheckTasks.err_res,
-                           check_items={ct.err_code: 65535,
-                                        ct.err_msg: f"cannot parse expression: {expression}, error: like "
-                                                    f"operation on non-string or no-json field is unsupported"})
+                           check_items={ct.err_code: 1100,
```
Comment (czs007) [inline tests/python_client/testcases/test_query.py:2514]: 2201

File: tests/python_client/testcases/test_query.py [hunk 7]
```diff
@@ -2542,8 +2541,7 @@ def test_query_compare_invalid_fields(self):
         expression = 'varchar == int64'
         collection_w.query(expression, check_task=CheckTasks.err_res,
                            check_items={ct.err_code: 1100, ct.err_msg:
```
Comment (czs007) [inline tests/python_client/testcases/test_query.py:2543]: 2201

**Comment by xiaocai2333** [comment]
return ut

**Comment by xiaofan-luan** [comment]
@xiaocai2333 
keep this for now

**Comment by xiaocai2333** [comment]
/repone
keep this

**Pull request closed without merging**